py-polars/polars/io.py

```diff
@@ -39,6 +39,13 @@
 except ImportError:
     _WITH_CX = False
 
+try:
+    import deltalake
```

Comment:
We have recently optimized our imports which means we cannot import any dependency anymore eagerly. See `py-polars/polars/dependencies.py` to see how we can deal with those.

You could also just import in the function as that is probably easier.

Reply:
importing in the function was causing some other problems with the return type of `_get_delta_lake_table` so I went with the lazy_imports.

If it makes sense, I can raise a separate MR to do this for `connectorx` as well (the `_WITH_CX` part)?

Reply:
Oh, yes makes sense. I think that connectorx missed our crosshairs. If you could do that, that would be great.

Reply:
alright, I will do it in a separate MR after this one.